No resource type found for:meta.k8s.io/v1#CreateOptions

There is no model for "CreateOptions" causing an exception while parsing the AdmissionReview.
The AdmissionReview object can be found in the official [kubernetes docs](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#webhook-request-and-response). 
This seems to be a problem for all cases (UpdateOptions, DeleteOptions) of the request->options portion of an AdmissionReview.

```java
  @Test
  public void testJacksonParsing() throws IOException {
    InputStream jsonStream = getClass().getResourceAsStream("/admissionReview.json");
    Serialization.unmarshal(jsonStream, AdmissionReview.class);
  }
```
### admissionReview.json :
```json
{
  "kind": "AdmissionReview",
  "apiVersion": "admission.k8s.io/v1beta1",
  "request": {
    "uid": "66105089-3471-46ff-b4c8-293546c2aa6c",
    "kind": {
      "group": "",
      "version": "v1",
      "kind": "Pod"
    },
    "resource": {
      "group": "",
      "version": "v1",
      "resource": "pods"
    },
    "requestKind": {
      "group": "",
      "version": "v1",
      "kind": "Pod"
    },
    "requestResource": {
      "group": "",
      "version": "v1",
      "resource": "pods"
    },
    "namespace": "myNamespace",
    "operation": "CREATE",
    "userInfo": {
      "username": "III#John.Doe@something.com",
      "groups": [
        "Group2"
      ]
    },
    "object": {
      "kind": "Pod",
      "apiVersion": "v1",
      "metadata": {
        "name": "example-pod",
        "namespace": "myNamespace",
        "creationTimestamp": null,
        "annotations": {
        }
      },
      "spec": {
        "volumes": [
        ],
        "containers": [

        ],
        "restartPolicy": "Always",
        "terminationGracePeriodSeconds": 30,
        "dnsPolicy": "ClusterFirst",
        "serviceAccountName": "default",
        "serviceAccount": "default",
        "securityContext": {},
        "schedulerName": "default-scheduler",
        "tolerations": [
          {
            "key": "node.kubernetes.io/not-ready",
            "operator": "Exists",
            "effect": "NoExecute",
            "tolerationSeconds": 600
          },
          {
            "key": "node.kubernetes.io/unreachable",
            "operator": "Exists",
            "effect": "NoExecute",
            "tolerationSeconds": 600
          }
        ],
        "priority": 0,
        "enableServiceLinks": true
      },
      "status": {}
    },
    "oldObject": null,
    "dryRun": false,
    "options": {
      "kind": "CreateOptions",
      "apiVersion": "meta.k8s.io/v1"
    }
  }
}
```

```
An error has occurred.
io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:64)
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:53)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:243)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:195)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:182)
	at EndpointTest.testJacksonParsing(EndpointTest.java:15)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        ...
Caused by: com.fasterxml.jackson.databind.JsonMappingException: No resource type found for:meta.k8s.io/v1#CreateOptions
 at [Source: (BufferedInputStream); line: 108, column: 5] (through reference chain: io.fabric8.kubernetes.api.model.admission.AdmissionReview["request"]->io.fabric8.kubernetes.api.model.admission.AdmissionRequest["options"])
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:271)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.fromObjectNode(KubernetesDeserializer.java:107)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:83)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:41)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)

```

--- a/kubernetes-model/kubernetes-model/pom.xml
+++ b/kubernetes-model/kubernetes-model/pom.xml
@@ -129,6 +129,7 @@
                 <delete file="${project.build.directory}/generated-sources/io/fabric8/kubernetes/api/model/Quantity.java" verbose="true" />
                 <delete file="${project.build.directory}/generated-sources/io/fabric8/kubernetes/api/model/WatchEvent.java" verbose="true" />
                 <delete file="${project.build.directory}/generated-sources/io/fabric8/openshift/api/model/Template.java" verbose="true" />
+                <delete file="${project.build.directory}/generated-sources/io/fabric8/kubernetes/api/model/admission/AdmissionRequest.java" verbose="true" />
                 <delete verbose="true">
                   <fileset dir="${project.build.directory}/generated-sources">
                     <include name="*.java" />

--- a/kubernetes-model/kubernetes-model/src/main/java/io/fabric8/kubernetes/api/model/admission/AdmissionRequest.java
+++ b/kubernetes-model/kubernetes-model/src/main/java/io/fabric8/kubernetes/api/model/admission/AdmissionRequest.java
@@ -1,0 +1,495 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.kubernetes.api.model.admission;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.fabric8.kubernetes.api.model.GroupVersionKind;
+import io.fabric8.kubernetes.api.model.GroupVersionResource;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+import io.fabric8.kubernetes.api.model.authentication.UserInfo;
+
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * AdmissionRequest describes the admission.Attributes for the admission request.
+ *
+ * This POJO is derived from https://github.com/kubernetes/api/blob/master/admission/v1beta1/types.go
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonDeserialize
+@ToString
+@EqualsAndHashCode
+public class AdmissionRequest implements KubernetesResource
+{
+
+  /*
+   * DryRun indicates that modifications will definitely not be persisted for this request.
+	 * Defaults to false.
+   */
+  @JsonProperty("dryRun")
+  private Boolean dryRun;
+
+  /*
+   * Kind is fully-qualified resource being requested (for example, v1.Pods)
+   */
+  @JsonProperty("kind")
+  private GroupVersionKind kind;
+
+  /*
+   * Name is the name of the object as presented in the request.  On a CREATE operation, the client may omit name and
+   * rely on the server to generate the name.  If that is the case, this field will contain an empty string.
+   */
+  @JsonProperty("name")
+  private String name;
+
+  /*
+   * Namespace is the namespace associated with the request (if any).
+   */
+  @JsonProperty("namespace")
+  private String namespace;
+
+  /*
+   * Object is the object from the incoming request.
+   */
+  @JsonProperty("object")
+  private HasMetadata object;
+
+  /*
+   * OldObject is the existing object. Only populated for DELETE and UPDATE requests.
+   */
+  @JsonProperty("oldObject")
+  private HasMetadata oldObject;
+
+  /*
+   * Operation is the operation being performed. This may be different than the operation
+	 * requested. e.g. a patch can result in either a CREATE or UPDATE Operation.
+   */
+  @JsonProperty("operation")
+  private String operation;
+
+  @JsonProperty("options")
+  private KubernetesResource options;
+
+  /*
+   * RequestResource is the fully-qualified resource of the original API request (for example, v1.pods).
+	 * If this is specified and differs from the value in "resource", an equivalent match and conversion was performed.
+	 *
+	 * For example, if deployments can be modified via apps/v1 and apps/v1beta1, and a webhook registered a rule of
+	 * `apiGroups:["apps"], apiVersions:["v1"], resources: ["deployments"]` and `matchPolicy: Equivalent`,
+	 * an API request to apps/v1beta1 deployments would be converted and sent to the webhook
+	 * with `resource: {group:"apps", version:"v1", resource:"deployments"}` (matching the resource the webhook registered for),
+	 * and `requestResource: {group:"apps", version:"v1beta1", resource:"deployments"}` (indicating the resource of the original API request).
+   */
+  @JsonProperty("requestKind")
+  private GroupVersionKind requestKind;
+
+  /*
+   * RequestResource is the fully-qualified resource of the original API request (for example, v1.pods).
+   * If this is specified and differs from the value in "resource", an equivalent match and conversion was performed.
+   *
+   * For example, if deployments can be modified via apps/v1 and apps/v1beta1, and a webhook registered a rule of
+   * `apiGroups:["apps"], apiVersions:["v1"], resources: ["deployments"]` and `matchPolicy: Equivalent`,
+   * an API request to apps/v1beta1 deployments would be converted and sent to the webhook
+   * with `resource: {group:"apps", version:"v1", resource:"deployments"}` (matching the resource the webhook registered for),
+   * and `requestResource: {group:"apps", version:"v1beta1", resource:"deployments"}` (indicating the resource of the original API request).
+   *
+   * See documentation for the "matchPolicy" field in the webhook configuration type.
+   */
+  @JsonProperty("requestResource")
+  private GroupVersionResource requestResource;
+
+  /*
+   * RequestSubResource is the name of the subresource of the original API request, if any (for example, "status" or "scale")
+   * If this is specified and differs from the value in "subResource", an equivalent match and conversion was performed.
+   * See documentation for the "matchPolicy" field in the webhook configuration type.
+   */
+  @JsonProperty("requestSubResource")
+  private String requestSubResource;
+
+  /*
+   * Resource is the fully-qualified resource being requested (for example, v1.pods)
+   */
+  @JsonProperty("resource")
+  private GroupVersionResource resource;
+
+  /*
+   * SubResource is the subresource being requested (for example, "status" or "scale")
+   */
+  @JsonProperty("subResource")
+  private String subResource;
+
+  /*
+   * UID is identifier for individual request/response. It allows us to distinguish instances of
+   * requests which otherwise identical (parallel requests, requests when earlier requests did not
+   * modify etc). The UID is meant to track the round trip (request/response) between KAS and
+   * the WebHook, not the user request.
+   * It is suitable for correlating log entries between the webhook and apiserver, for either
+   * auditing or debugging.
+   */
+  @JsonProperty("uid")
+  private String uid;
+
+  /*
+   * UserInfo is information about the requesting user
+   */
+  @JsonProperty("userInfo")
+  private UserInfo userInfo;
+
+  @JsonIgnore
+  private Map<String, Object> additionalProperties = new HashMap<>();
+
+  public AdmissionRequest() {
+  }
+
+  public AdmissionRequest(Boolean dryRun, GroupVersionKind kind, String name, String namespace, HasMetadata object, HasMetadata oldObject, String operation, HasMetadata options, GroupVersionKind requestKind, GroupVersionResource requestResource, String requestSubResource, GroupVersionResource resource, String subResource, String uid, UserInfo userInfo) {
+    this.dryRun = dryRun;
+    this.kind = kind;
+    this.name = name;
+    this.namespace = namespace;
+    this.object = object;
+    this.oldObject = oldObject;
+    this.operation = operation;
+    this.options = options;
+    this.requestKind = requestKind;
+    this.requestResource = requestResource;
+    this.requestSubResource = requestSubResource;
+    this.resource = resource;
+    this.subResource = subResource;
+    this.uid = uid;
+    this.userInfo = userInfo;
+  }
+
+  /**
+   * Get Dry Run
+   *
+   * @return The dryRun
+   */
+  @JsonProperty("dryRun")
+  public Boolean getDryRun() {
+    return dryRun;
+  }
+
+  /**
+   * Set Dry run
+   * @param dryRun The dryRun
+   */
+  @JsonProperty("dryRun")
+  public void setDryRun(Boolean dryRun) {
+    this.dryRun = dryRun;
+  }
+
+  /**
+   * Get Kind
+   *
+   * @return The kind
+   */
+  @JsonProperty("kind")
+  public GroupVersionKind getKind() {
+    return kind;
+  }
+
+  /**
+   * Set Kind
+   *
+   * @param kind The kind
+   */
+  @JsonProperty("kind")
+  public void setKind(GroupVersionKind kind) {
+    this.kind = kind;
+  }
+
+  /**
+   * Get Name
+   *
+   * @return The name
+   */
+  @JsonProperty("name")
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * Set Name
+   *
+   * @param name The name
+   */
+  @JsonProperty("name")
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  /**
+   * Get Namespace
+   *
+   * @return The namespace
+   */
+  @JsonProperty("namespace")
+  public String getNamespace() {
+    return namespace;
+  }
+
+  /**
+   * Set Namespace
+   *
+   * @param namespace The namespace
+   */
+  @JsonProperty("namespace")
+  public void setNamespace(String namespace) {
+    this.namespace = namespace;
+  }
+
+  /**
+   * Get Object
+   *
+   * @return The object
+   */
+  @JsonProperty("object")
+  public HasMetadata getObject() {
+    return object;
+  }
+
+  /**
+   * Set Object
+   *
+   * @param object The object
+   */
+  @JsonProperty("object")
+  public void setObject(HasMetadata object) {
+    this.object = object;
+  }
+
+  /**
+   * Get old object.
+   *
+   * @return The oldObject
+   */
+  @JsonProperty("oldObject")
+  public HasMetadata getOldObject() {
+    return oldObject;
+  }
+
+  /**
+   * Set old object.
+   *
+   * @param oldObject The oldObject
+   */
+  @JsonProperty("oldObject")
+  public void setOldObject(HasMetadata oldObject) {
+    this.oldObject = oldObject;
+  }
+
+  /**
+   * Get operation.
+   *
+   * @return The operation
+   */
+  @JsonProperty("operation")
+  public String getOperation() {
+    return operation;
+  }
+
+  /**
+   * Set operation.
+   *
+   * @param operation The operation
+   */
+  @JsonProperty("operation")
+  public void setOperation(String operation) {
+    this.operation = operation;
+  }
+
+  /**
+   * Get options.
+   *
+   * @return The options
+   */
+  @JsonProperty("options")
+  public KubernetesResource getOptions() {
+    return options;
+  }
+
+  /**
+   * Set options.
+   *
+   * @param options The options
+   */
+  @JsonProperty("options")
+  public void setOptions(KubernetesResource options) {
+    this.options = options;
+  }
+
+  /**
+   * Get Request Kind.
+   *
+   * @return The requestKind
+   */
+  @JsonProperty("requestKind")
+  public GroupVersionKind getRequestKind() {
+    return requestKind;
+  }
+
+  /**
+   * Set RequestKind
+   *
+   * @param requestKind The requestKind
+   */
+  @JsonProperty("requestKind")
+  public void setRequestKind(GroupVersionKind requestKind) {
+    this.requestKind = requestKind;
+  }
+
+  /**
+   * Get RequestResource
+   *
+   * @return The requestResource
+   */
+  @JsonProperty("requestResource")
+  public GroupVersionResource getRequestResource() {
+    return requestResource;
+  }
+
+  /**
+   * Set RequestResource
+   *
+   * @param requestResource The requestResource
+   */
+  @JsonProperty("requestResource")
+  public void setRequestResource(GroupVersionResource requestResource) {
+    this.requestResource = requestResource;
+  }
+
+  /**
+   * Get RequestSubResource
+   *
+   * @return The requestSubResource
+   */
+  @JsonProperty("requestSubResource")
+  public String getRequestSubResource() {
+    return requestSubResource;
+  }
+
+  /**
+   * Set RequestSubResource
+   *
+   * @param requestSubResource The requestSubResource
+   */
+  @JsonProperty("requestSubResource")
+  public void setRequestSubResource(String requestSubResource) {
+    this.requestSubResource = requestSubResource;
+  }
+
+  /**
+   * Get Resource
+   *
+   * @return The resource
+   */
+  @JsonProperty("resource")
+  public GroupVersionResource getResource() {
+    return resource;
+  }
+
+  /**
+   * Set Resource.
+   *
+   * @param resource The resource
+   */
+  @JsonProperty("resource")
+  public void setResource(GroupVersionResource resource) {
+    this.resource = resource;
+  }
+
+  /**
+   * Get SubResource
+   *
+   * @return The subResource
+   */
+  @JsonProperty("subResource")
+  public String getSubResource() {
+    return subResource;
+  }
+
+  /**
+   * Set SubResource
+   *
+   * @param subResource The subResource
+   */
+  @JsonProperty("subResource")
+  public void setSubResource(String subResource) {
+    this.subResource = subResource;
+  }
+
+  /**
+   * Get Uid
+   *
+   * @return The uid
+   */
+  @JsonProperty("uid")
+  public String getUid() {
+    return uid;
+  }
+
+  /**
+   * Set Uid
+   *
+   * @param uid The uid
+   */
+  @JsonProperty("uid")
+  public void setUid(String uid) {
+    this.uid = uid;
+  }
+
+  /**
+   * Get UserInfo
+   *
+   * @return The userInfo
+   */
+  @JsonProperty("userInfo")
+  public UserInfo getUserInfo() {
+    return userInfo;
+  }
+
+  /**
+   * Set UserInfo
+   *
+   * @param userInfo The userInfo
+   */
+  @JsonProperty("userInfo")
+  public void setUserInfo(UserInfo userInfo) {
+    this.userInfo = userInfo;
+  }
+
+  @JsonAnyGetter
+  public Map<String, Object> getAdditionalProperties() {
+    return this.additionalProperties;
+  }
+
+  @JsonAnySetter
+  public void setAdditionalProperty(String name, Object value) {
+    this.additionalProperties.put(name, value);
+  }
+
+}

--- a/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/AdmissionReviewTest.java
+++ b/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/AdmissionReviewTest.java
@@ -1,0 +1,63 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.mock;
+
+import io.fabric8.kubernetes.api.model.admission.AdmissionReview;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class AdmissionReviewTest {
+  @Test
+  @DisplayName("Should be able to deserialize from AdmissionRequest option set to CreateOption")
+  public void testJacksonParsingWithCreateOptions() throws IOException {
+    InputStream jsonStream = getClass().getResourceAsStream("/admissionreview.json");
+    AdmissionReview admissionReview = Serialization.unmarshal(jsonStream, AdmissionReview.class);
+    assertEquals("CREATE", admissionReview.getRequest().getOperation());
+    assertRequest(admissionReview);
+  }
+
+  @Test
+  @DisplayName("Should be able to deserialize from AdmissionRequest option set to UpdateOption")
+  public void testJacksonParsingWithUpdateOptions() throws IOException {
+    InputStream jsonStream = getClass().getResourceAsStream("/admissionreview-withupdateoptions.json");
+    AdmissionReview admissionReview = Serialization.unmarshal(jsonStream, AdmissionReview.class);
+    assertEquals("UPDATE", admissionReview.getRequest().getOperation());
+    assertRequest(admissionReview);
+  }
+
+  @Test
+  @DisplayName("Should be able to deserialize from AdmissionRequest option set to PatchOption")
+  public void testJacksonParsingWithPathOptions() throws IOException {
+    InputStream jsonStream = getClass().getResourceAsStream("/admissionreview-withpatchoptions.json");
+    AdmissionReview admissionReview = Serialization.unmarshal(jsonStream, AdmissionReview.class);
+    assertEquals("PATCH", admissionReview.getRequest().getOperation());
+    assertRequest(admissionReview);
+  }
+
+  private void assertRequest(AdmissionReview admissionReview) {
+    assertEquals("myNamespace", admissionReview.getRequest().getNamespace());
+    assertEquals("66105089-3471-46ff-b4c8-293546c2aa6c", admissionReview.getRequest().getUid());
+    assertEquals("III#John.Doe@something.com", admissionReview.getRequest().getUserInfo().getUsername());
+    assertEquals(Collections.singletonList("Group2"), admissionReview.getRequest().getUserInfo().getGroups());
+  }
+}

--- a/kubernetes-tests/src/test/resources/admissionreview-withpatchoptions.json
+++ b/kubernetes-tests/src/test/resources/admissionreview-withpatchoptions.json
@@ -1,0 +1,83 @@
+{
+  "kind": "AdmissionReview",
+  "apiVersion": "admission.k8s.io/v1beta1",
+  "request": {
+    "uid": "66105089-3471-46ff-b4c8-293546c2aa6c",
+    "kind": {
+      "group": "",
+      "version": "v1",
+      "kind": "Pod"
+    },
+    "resource": {
+      "group": "",
+      "version": "v1",
+      "resource": "pods"
+    },
+    "requestKind": {
+      "group": "",
+      "version": "v1",
+      "kind": "Pod"
+    },
+    "requestResource": {
+      "group": "",
+      "version": "v1",
+      "resource": "pods"
+    },
+    "namespace": "myNamespace",
+    "operation": "PATCH",
+    "userInfo": {
+      "username": "III#John.Doe@something.com",
+      "groups": [
+        "Group2"
+      ]
+    },
+    "object": {
+      "kind": "Pod",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "example-pod",
+        "namespace": "myNamespace",
+        "creationTimestamp": null,
+        "annotations": {
+        }
+      },
+      "spec": {
+        "volumes": [
+        ],
+        "containers": [
+
+        ],
+        "restartPolicy": "Always",
+        "terminationGracePeriodSeconds": 30,
+        "dnsPolicy": "ClusterFirst",
+        "serviceAccountName": "default",
+        "serviceAccount": "default",
+        "securityContext": {},
+        "schedulerName": "default-scheduler",
+        "tolerations": [
+          {
+            "key": "node.kubernetes.io/not-ready",
+            "operator": "Exists",
+            "effect": "NoExecute",
+            "tolerationSeconds": 600
+          },
+          {
+            "key": "node.kubernetes.io/unreachable",
+            "operator": "Exists",
+            "effect": "NoExecute",
+            "tolerationSeconds": 600
+          }
+        ],
+        "priority": 0,
+        "enableServiceLinks": true
+      },
+      "status": {}
+    },
+    "oldObject": null,
+    "dryRun": false,
+    "options": {
+      "kind": "PatchOptions",
+      "apiVersion": "meta.k8s.io/v1"
+    }
+  }
+}

--- a/kubernetes-tests/src/test/resources/admissionreview-withupdateoptions.json
+++ b/kubernetes-tests/src/test/resources/admissionreview-withupdateoptions.json
@@ -1,0 +1,83 @@
+{
+  "kind": "AdmissionReview",
+  "apiVersion": "admission.k8s.io/v1beta1",
+  "request": {
+    "uid": "66105089-3471-46ff-b4c8-293546c2aa6c",
+    "kind": {
+      "group": "",
+      "version": "v1",
+      "kind": "Pod"
+    },
+    "resource": {
+      "group": "",
+      "version": "v1",
+      "resource": "pods"
+    },
+    "requestKind": {
+      "group": "",
+      "version": "v1",
+      "kind": "Pod"
+    },
+    "requestResource": {
+      "group": "",
+      "version": "v1",
+      "resource": "pods"
+    },
+    "namespace": "myNamespace",
+    "operation": "UPDATE",
+    "userInfo": {
+      "username": "III#John.Doe@something.com",
+      "groups": [
+        "Group2"
+      ]
+    },
+    "object": {
+      "kind": "Pod",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "example-pod",
+        "namespace": "myNamespace",
+        "creationTimestamp": null,
+        "annotations": {
+        }
+      },
+      "spec": {
+        "volumes": [
+        ],
+        "containers": [
+
+        ],
+        "restartPolicy": "Always",
+        "terminationGracePeriodSeconds": 30,
+        "dnsPolicy": "ClusterFirst",
+        "serviceAccountName": "default",
+        "serviceAccount": "default",
+        "securityContext": {},
+        "schedulerName": "default-scheduler",
+        "tolerations": [
+          {
+            "key": "node.kubernetes.io/not-ready",
+            "operator": "Exists",
+            "effect": "NoExecute",
+            "tolerationSeconds": 600
+          },
+          {
+            "key": "node.kubernetes.io/unreachable",
+            "operator": "Exists",
+            "effect": "NoExecute",
+            "tolerationSeconds": 600
+          }
+        ],
+        "priority": 0,
+        "enableServiceLinks": true
+      },
+      "status": {}
+    },
+    "oldObject": null,
+    "dryRun": false,
+    "options": {
+      "kind": "UpdateOptions",
+      "apiVersion": "meta.k8s.io/v1"
+    }
+  }
+}

--- a/kubernetes-tests/src/test/resources/admissionreview.json
+++ b/kubernetes-tests/src/test/resources/admissionreview.json
@@ -1,0 +1,83 @@
+{
+  "kind": "AdmissionReview",
+  "apiVersion": "admission.k8s.io/v1beta1",
+  "request": {
+    "uid": "66105089-3471-46ff-b4c8-293546c2aa6c",
+    "kind": {
+      "group": "",
+      "version": "v1",
+      "kind": "Pod"
+    },
+    "resource": {
+      "group": "",
+      "version": "v1",
+      "resource": "pods"
+    },
+    "requestKind": {
+      "group": "",
+      "version": "v1",
+      "kind": "Pod"
+    },
+    "requestResource": {
+      "group": "",
+      "version": "v1",
+      "resource": "pods"
+    },
+    "namespace": "myNamespace",
+    "operation": "CREATE",
+    "userInfo": {
+      "username": "III#John.Doe@something.com",
+      "groups": [
+        "Group2"
+      ]
+    },
+    "object": {
+      "kind": "Pod",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "example-pod",
+        "namespace": "myNamespace",
+        "creationTimestamp": null,
+        "annotations": {
+        }
+      },
+      "spec": {
+        "volumes": [
+        ],
+        "containers": [
+
+        ],
+        "restartPolicy": "Always",
+        "terminationGracePeriodSeconds": 30,
+        "dnsPolicy": "ClusterFirst",
+        "serviceAccountName": "default",
+        "serviceAccount": "default",
+        "securityContext": {},
+        "schedulerName": "default-scheduler",
+        "tolerations": [
+          {
+            "key": "node.kubernetes.io/not-ready",
+            "operator": "Exists",
+            "effect": "NoExecute",
+            "tolerationSeconds": 600
+          },
+          {
+            "key": "node.kubernetes.io/unreachable",
+            "operator": "Exists",
+            "effect": "NoExecute",
+            "tolerationSeconds": 600
+          }
+        ],
+        "priority": 0,
+        "enableServiceLinks": true
+      },
+      "status": {}
+    },
+    "oldObject": null,
+    "dryRun": false,
+    "options": {
+      "kind": "CreateOptions",
+      "apiVersion": "meta.k8s.io/v1"
+    }
+  }
+}

